sdk/iot/core/inc/az_iot_hub_client.h

```diff
@@ -58,7 +58,7 @@ AZ_NODISCARD az_iot_hub_client_options az_iot_hub_client_options_default();
  * @param[out] client The #az_iot_hub_client to use for this call.
  * @param[in] iot_hub_hostname The IoT Hub Hostname.
  * @param[in] device_id The Device ID.
- * @param[in] options A reference to an #az_iot_hub_client_options structure. Can be NULL.
+ * @param[in] options A reference to an #az_iot_hub_client_options structure. Cannot be `NULL`.

```

Comment:
I don't think we need the last sentence. how about

 The #az_iot_hub_client_options to use for this call.

Reply:
@jspaith, @JeffreyRichter and I talked about this during a review meeting and I remember the pattern was that NULL can be passed. In that case, internally the defaults will be called.

https://azure.github.io/azure-sdk/clang_design.html#functions-with-many-parameters

Reply:
I think the documentation should detail the expectation for the parameters. Otherwise users have to look at the implementation in `.c` files to understand what the preconditions are for functions.